### PR TITLE
Auth service

### PR DESCRIPTION
**Pull request by leafchild0**
Auth service. Has all the security stuff in it, DB integration, table for users. Login and register operations.
Also API for users

### CODE REVIEW

File: auth/src/main/resources/data/schema.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+drop table IF EXISTS users;
```
Comment (proteus1121) [inline auth/src/main/resources/data/schema.sql:1]: I use flyway for db init, maybe make sense to use it for users also, for painless adding some additional fields in the future?

File: auth/src/main/java/com/chartdesk/auth/security/UserPrincipal.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		if (o == null || getClass() != o.getClass()) return false;
+		UserPrincipal that = (UserPrincipal) o;
+		return Objects.equals(id, that.id);
+	}
+
+	@Override
+	public int hashCode()
+	{
+		return Objects.hash(id);
+	}
```
[the remark this answers is not in the document]
Comment (leafchild0) [reply auth/src/main/java/com/chartdesk/auth/security/UserPrincipal.java]: Done

File: auth/src/main/resources/data/schema.sql (hunk 1 above)
Comment (leafchild0) [reply auth/src/main/resources/data/schema.sql]: Added

File: auth/src/main/resources/bootstrap.yml [hunk 3]
```diff
@@ -16,6 +16,12 @@ spring:
     platform: postgres
     initialization-mode: always
     continue-on-error: true
+    schema: users
+  flyway:
+    schemas: users
+    url: jdbc:postgresql://localhost/chartdesk
```
Comment (proteus1121) [inline auth/src/main/resources/bootstrap.yml:22]: I think DB should be 'users' or 'auth'?

**Review by proteus1121 — APPROVED** [review]

**Pull request merged**